File: Tutorials and Problems/P2d/Solutions d/LunchList4/app/src/main/java/org/feup/apm/lunchlist4/DetailsActivity.java
package org.feup.apm.lunchlist4;

import android.database.Cursor;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioGroup;

public class DetailsActivity extends AppCompatActivity {
  RestaurantsHelper helper;
  String rId;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    ActionBar bar = getSupportActionBar();
    if (bar != null) {
      bar.setIcon(R.drawable.rest_icon);
      bar.setDisplayShowHomeEnabled(true);
    }
    setContentView(R.layout.activity_details);
    helper = new RestaurantsHelper(this);
    findViewById(R.id.bt_save).setOnClickListener((View v)->onBtSaveClick());

    rId=getIntent().getStringExtra(MainActivity.ID_EXTRA);
    if (rId != null)
      load();
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    helper.close();
  }

  void onBtSaveClick() {
    String type = "";

    switch (((RadioGroup)findViewById(R.id.rg_types)).getCheckedRadioButtonId()) {
      case R.id.sit:
        type = "sit";
        break;
      case R.id.take:
        type = "take";
        break;
      case R.id.delivery:
        type = "delivery";
        break;
    }
    if (rId==null) {
      MainActivity.currentId = helper.insert(
          ((EditText)findViewById(R.id.ed_name)).getText().toString(),
          ((EditText)findViewById(R.id.ed_address)).getText().toString(),
          type,
          ((EditText)findViewById(R.id.ed_notes)).getText().toString() );
    }
    else {
      helper.update(rId,
          ((EditText)findViewById(R.id.ed_name)).getText().toString(),
          ((EditText)findViewById(R.id.ed_address)).getText().toString(),
          type,
          ((EditText)findViewById(R.id.ed_notes)).getText().toString() );
    }
    finish();
  }

  void load() {
    Cursor c=helper.getById(rId);
    c.moveToFirst();
    ((EditText)findViewById(R.id.ed_name)).setText(helper.getName(c));
    ((EditText)findViewById(R.id.ed_address)).setText(helper.getAddress(c));
    ((EditText)findViewById(R.id.ed_notes)).setText(helper.getNotes(c));
    RadioGroup rgTypes = findViewById(R.id.rg_types);
    if (helper.getType(c).equals("sit"))
      rgTypes.check(R.id.sit);
    else if (helper.getType(c).equals("take"))
      rgTypes.check(R.id.take);
    else
      rgTypes.check(R.id.delivery);
    c.close();
  }
}

File: Tutorials and Problems/3.4. The Lunch Restaurants App/LunchList4/app/src/main/java/org/feup/apm/lunchlist4/DetailsActivity.kt
package org.feup.apm.lunchlist4

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class DetailsActivity : AppCompatActivity() {
  private val dbHelper by lazy { RestaurantHelper(this) }
  private val rId: String? by lazy { intent.getStringExtra(ID_EXTRA) }
  private val edName by lazy { findViewById<EditText>(R.id.ed_name) }
  private val edAddress by lazy { findViewById<EditText>(R.id.ed_address) }
  private val edNotes by lazy { findViewById<EditText>(R.id.ed_notes) }
  private val rgTypes by lazy { findViewById<RadioGroup>(R.id.rg_types) }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    supportActionBar?.setIcon(R.drawable.rest_icon)
    supportActionBar?.setDisplayShowHomeEnabled(true)
    setContentView(R.layout.activity_details)

    findViewById<Button>(R.id.bt_save).setOnClickListener { _ -> onBtSaveClick() }
    if (rId != null) load()
  }

  override fun onDestroy() {
    super.onDestroy()
    dbHelper.close()
  }

  private fun onBtSaveClick() {
    val rName = edName.text.toString()
    val rAddress = edAddress.text.toString()
    val rNotes = edNotes.text.toString()
    val rType = when (rgTypes.checkedRadioButtonId) {
      R.id.sit -> "sit"
      R.id.take -> "take"
      R.id.delivery -> "delivery"
      else -> ""
    }
    if (rId == null)
      currentId = dbHelper.insert(rName, rAddress, rType, rNotes)
    else
      dbHelper.update(rId!!, rName, rAddress, rType, rNotes)
    finish()
  }

  private fun load() {
    val c = dbHelper.getById(rId!!)
    c.moveToFirst()
    edName.setText(dbHelper.getName(c))
    edAddress.setText(dbHelper.getAddress(c))
    edNotes.setText(dbHelper.getNotes(c))
    when (dbHelper.getType(c)) {
      "sit" -> rgTypes.check(R.id.sit)
      "take" -> rgTypes.check(R.id.take)
      "delivery" -> rgTypes.check(R.id.delivery)
    }
    c.close()
  }
}